DIAGRAMS.md

```diff
@@ -0,0 +1,304 @@
+# Diagrams
+
+# Project Structure
+Kyoo is a monorepo that consists of several projects each in their own directory.  Diagram below shows an outline of kyoo, projects, and artifacts.
+
+```mermaid
+block-beta
+  columns 1
+  block:proj1:1
+    proj_name["Kyoo"]:1
+  end
+  block:proj2:1
+    dir_1["autosync/"]
+    dir_2["back/"]
+    dir_3["front/"]
+    dir_4["transcoder/"]
+    dir_5["scanner/"]
+  end
+  block:proj3:1
+    %% columns auto (default)
+    block:autosync_b:1
+      autosync_i1("kyoo_autosync")
+    end
+    block:back_b:1
+      columns 1
+      back_i1("kyoo_back")
+      back_i2("kyoo_migrations")
+    end
+    block:front_b:1
+      front_i1("kyoo_front")
+    end
+    block:transcoder_b:1
+      transcoder_i1("kyoo_transcoder")
+    end
+    block:scanner_b:1
+      columns 1
+      scanner_i1("kyoo_scanner")
+      scanner_i2("kyoo_scanner*")
```

Comment:
we should specify somehow that this is the matcher, don't you think?

Reply:
I definitely agree.  Lets create a matcher image and then update this diagram. These diagrams should be as close to what is actually deployed.

Ideally we should not be required specify docker args at deployment time.

Reply:
I don't think having a copy of an image with only the flag changing is a good way.

Even grapahana does that with a flag for reader/writter (I copied their way when doing that).